realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -492,6 +497,54 @@ public long getVersion() {
         return OsObjectStore.getSchemaVersion(sharedRealm);
     }
 
+    /**
+     * Returns the privileges granted to the current user for this Realm.
+     *
+     * @return the privileges granted the current user for this Realm.
+     */
+    @Beta
+    abstract public RealmPrivileges getPrivileges();
+
+    /**
+     * Returns the privileges granted to the current user for the given object.
+     *
+     * @param object Realm object to get privileges for.
+     * @return the privileges granted the current user for the object.
+     * @throws IllegalArgumentException if the object is either unmanaged or not part of this Realm.
+     */
+    @Beta
+    abstract public RealmPrivileges getPrivileges(RealmModel object);
+
+    /**
+     * Returns all permissions associated with the current Realm. Attach a change listener
+     * using {@link RealmPermissions#addChangeListener(RealmChangeListener)} to be notified about
+     * any future changes.
+     *
+     * @return all permissions for the current Realm.
+     */
+    @Beta
+    abstract public RealmPermissions getPermissions();
+
+    /**
+     * Returns all {@link Role} objects available in this Realm. Attach a change listener
+     * using {@link Role#addChangeListener(RealmChangeListener)} to be notified about
+     * any future changes.
+     *
+     * @return all roles available in the current Realm.
+     */
+    @Beta
+    abstract public RealmResults<Role> getRoles();
+
+    /**
+     * Returns all {@link User} objects available in this Realm. Attach a change listener
+     * using {@link User#addChangeListener(RealmChangeListener)} to be notified about
+     * any future changes.
+     *
+     * @return all roles available in the current Realm.
+     */
+    @Beta
+    abstract public RealmResults<User> getUsers();
```

Comment:
I suggest we work only with "user_id" as a string, this eliminates the need for the conflicting `User` class, 
example
```Java
realm.getRoles().where("name", "my-role").findFirst().getMembers().add(new User(syncUser.getIdentity()));
```
becomes 
```Java
realm.getRoles().where("name", "my-role").findFirst().getMembers().add(syncUser.getIdentity());
```